src/walletd/runtime.rs

```diff
@@ -1382,6 +1544,56 @@ impl Runtime {
     }
 }
 
+pub fn checkpoint_state(
+    endpoints: &mut Endpoints,
+    swap_id: SwapId,
+    state: request::CheckpointState,
+) -> Result<(), Error> {
+    let mut serialized_state = vec![];
+    let size = state.strict_encode(&mut serialized_state).unwrap();
+
+    // if the size exceeds a boundary, send a multi-part message
+    let max_chunk_size = internet2::transport::MAX_FRAME_SIZE - 1024;
```

Comment:
I find the the client daemons (here wallet) to have to multipart messages not the correct api, but i guess if this code was in databased, it would feel more natural, its just an import for the client daemon